src/initsout.c

```diff
@@ -140,9 +140,9 @@ void init_ifpage(unsigned sysout_size) {
 #endif /* BIGVM */
 
   /* unfortunately, Lisp only looks at a 16 bit serial number */
-#ifndef DOS
+#if !defined(DOS) && !defined(MAIKO_OS_HAIKU)
   InterfacePage->serialnumber = 0xffff & gethostid();
```

Comment:
I'm surprised Haiku doesn't have a `gethostid()` since the [libroot documentation](https://www.haiku-os.org/docs/api/libroot.html) says it implements [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/) calls which include `gethostid()`

Reply:
@nbriggs https://dev.haiku-os.org/ticket/10841

Reply:
We should also open an issue against the libroot documentation to suggest that it include a list of unimplemented features rather than just referring to the POSIX spec and then not implementing some of it.  That would have shortened the journey to this answer.

The gethostid() call used to (on Sun hardware) return the hardware serial number.  The underlying implementation changed, but fairly early on it moved from its own system call to be a compatibility function loosely based on the (SunOS/Solaris) sysinfo() call, SI_HW_SERIAL, result
>       SI_HW_SERIAL

           Copy  into  the array pointed to by buf a string which is the ASCII
           representation of the hardware-specific serial number of the physi-
           cal machine on which the function is executed. This might be imple-
           mented in Read-Only  Memory,  using  software  constants  set  when
           building the operating system, or by other means, and might contain
           non-numeric characters. If the function is executed within  a  non-
           global  zone that emulates a host identifier, then the ASCII repre-
           sentation of the zone's host identifier is copied  into  the  array
           pointed  to  by  buf. It is anticipated that manufacturers will not
           issue the same "serial number" to more than one  physical  machine.
           The  pair of strings returned by SI_HW_PROVIDER and SI_HW_SERIAL is
           not guaranteed to be unique across all  vendor's  SVR4  implementa-
           tions and could change over the lifetime of a given system.

I'm not sure what glibc implementors were thinking when they based it on the IP address - that would never have been a reasonable thing to do

Reply:
A point to note in all this, though, is that the only place the "hostid" now shows up is in the common lisp environment code where it effectively tries to do a gethostid() and if that fails it tries for HOSTID (as a string) in the Unix environment.  Maiko used to be licensed software, based on the hostid, but that is long gone.  The right thing would be to use the system-uuid, but there's no standardized (and unprivileged) way of determining that on the variety of systems that we run on.

Reply:
I don't think there is any unique identifier one can retrieve for a modern computer that's likely to be both accurate and privacy-respecting.

Emacs tries (or tried) to infer the user's email address based on the hostname and username and that often failed. Peronal computers don't tend to have a hostname that makes sense.

Some device ID (e.g. Ethernet MAC address) would probably be the most reliable in practice, but poking around hardware details like that is behavior that is nowadays mainly attributed to malware.

Reply:
How about making it a variable whose default value is either zero or a random number?

Reply:
People who care about it could assign a value of their choice. Perhaps in a config file.

Reply:
There is code in multiplayer LOOPS that I remember seeing that relies on stable hostid across savevm and restore (but not sysout or makesys presumably)

Reply:
@masinter - hmm.  I see truckin/LOOPSGATEWAY setting up `(SETQ \MY.NSHOSTID (CONCAT "#NSID." (CADR \MY.NSHOSTNUMBER)` but I don't see any direct reference to the InterfacePage serial number, or to the Unix getenv/getparm (which would be unexpected since it wouldn't rely on there being a Unix system underneath, right?).

Reply:
Reproducing the full comment from the Haiku bug tracker:

> - The implementation in glibc attempts to compute an hostid based on the local IP, but on Debian it computes it based on 127.0.0.1 instead so all machines have the same host id.
> - musl just returns 0
> - NetBSD, FreeBSD and OpenBSD implements it but mentions that is deprecated and that it had been removed in 4.4BSD 

Note that some Linux distributions (e.g. Alpine Linux) use `musl` as the default C library. And of course Debian is not an unpopular distribution...

Reply:
@nbriggs I didn't remember the details, I just flagged this as a possible reason why Truckin didn't find player names... that unique IDs weren't necessarily unique?
https://github.com/Interlisp/medley/issues/260 has more details.

Reply:
As per [https://dev.haiku-os.org/ticket/10841](https://dev.haiku-os.org/ticket/10841), Haiku won't implement gethostid(), so i advise to stick as the same behaviour as DOS.